backend/services/organization.py

```diff
@@ -0,0 +1,185 @@
+__copyright__ = "Copyright 2023"
+__license__ = "MIT"
+
+class OrganizationService:
+    """Service that performs all of the actions on the `Organization` table"""
+
+    # Current SQLAlchemy Session
+    _session: Session
+
+    def __init__(
+        self,
+        session: Session = Depends(db_session),
+        permission: PermissionService = Depends(),
+    ):
+        """Initializes the `OrganizationService` session, and `PermissionService`"""
+        self._session = session
+        self._permission = permission
+
+    def all(self) -> list[Organization]:
+        """
+        Retrieves all organizations from the table
+
+        Returns:
+            list[Organization]: List of all `Organization`
+        """
+        # Select all entries in `Organization` table
+        query = select(OrganizationEntity)
+        entities = self._session.scalars(query).all()
+
+        # Convert entries to a model and return
+        return [entity.to_model() for entity in entities]
+
+    def create(self, subject: User, organization: Organization) -> Organization:
+        """
+        Creates a organization based on the input object and adds it to the table.
+        If the organization's ID is unique to the table, a new entry is added.
+        If the organization's ID already exists in the table, it raises an error.
+
+        Parameters:
+            subject: a valid User model representing the currently logged in User
+            organization (Organization): Organization to add to table
+
+        Returns:
+            Organization: Object added to table
+        """
+
+        # Check if user has admin permissions
+        self._permission.enforce(subject, "organization.create", f"organization")
+
+        # Checks if the organization already exists in the table
+        if organization.id:
+            # If so, raise an error
+            raise Exception(f"Duplicate organization found with ID: {organization.id}")
+
+        else:
+            # Otherwise, create new object
+            organization_entity = OrganizationEntity.from_model(organization)
+
+            # Add new object to table and commit changes
+            self._session.add(organization_entity)
+            self._session.commit()
+
+            # Return added object
+            return organization_entity.to_model()
+
+    def get_from_slug(self, slug: str) -> Organization:
+        """
+        Get the organization from a slug
+        If none retrieved, a debug description is displayed.
+
+        Parameters:
+            slug: a string representing a unique organization slug
+
+        Returns:
+            Organization: Object with corresponding slug
+
+        Raises:
+            Exception if no organization is found with the corresponding slug
+        """
+
+        # Query the organization with matching slug
+        organization = self._session.query(OrganizationEntity).filter(
+            OrganizationEntity.slug == slug
+        )[0]
```

Comment:
As a note for us later, I think another way of writing this would be: `organization = self._session.query(OrganizationEntity).filter(OrganizationEntity.slug == slug).one_or_none()` !

Reply:
Also potentially explore:
```py
session.execute(
    select(user).
    filter_by(name="Foo")
).scalar_one()
```